discovery_engine/lib/src/domain/feed_manager.dart

```diff
@@ -109,6 +111,31 @@ class FeedManager {
 
   /// Obtain the next batch of feed documents and persist to repositories.
   Future<EngineEvent> nextFeedBatch() async {
+    if (cfgFeatureStorage) {
+      final sources = await _sourceReactedRepo.fetchAll();
+      final List<DocumentWithActiveData> docs;
```

Comment:
might be better to mark it as `late final List<DocumentWithActiveData> docs;`, but if the analyzer doesn't complain, then maybe that's fine too 🤔

Reply:
if i understood this correctly, then `late final` does the check for initialization at runtime, whereas just declaring the variable (as `final`, and to have it available in the outer scope) and then assigning it in every if-branch (or returning early) does the check at compile time.